examples/networking/dns_matching/dns_matching.py

```diff
@@ -46,12 +52,34 @@ def encode_dns(name):
 # Get the table.
 cache = bpf.get_table("cache")
 
-# Create first entry for foo.bar
-key = cache.Key()
-key.p = encode_dns("foo.bar")
+# Add cache entries
+add_cache_entry(cache, "foo.bar")
+add_cache_entry(cache, "another.sample.domain")
+
+socket_fd = function_dns_matching.sock
+sock = socket.fromfd(socket_fd, socket.PF_PACKET, socket.SOCK_RAW, socket.IPPROTO_IP)
+sock.setblocking(True)
+
+while 1:
+  #retrieve raw packet from socket
+  packet_str = os.read(socket_fd, 2048)
+  packet_bytearray = bytearray(packet_str)
+
+  ETH_HLEN = 14
+  UDP_HLEN = 8
+
+  #IP HEADER
+  #calculate ip header length
+  ip_header_length = packet_bytearray[ETH_HLEN]               #load Byte
+  ip_header_length = ip_header_length & 0x0F                  #mask bits 0..3
+  ip_header_length = ip_header_length << 2                    #shift to obtain length
+
+  #calculate payload offset
+  payload_offset = ETH_HLEN + ip_header_length + UDP_HLEN
```

Comment:
How do you know it must be UDP packet?

Reply:
Because bpf program is filtering-in only UDP packets (only those containing DNS question).

Reply:
Oh, right. the bpf program drops everything except dns packets.